libyul/backends/evm/StackLayoutGenerator.cpp

```diff
@@ -540,6 +594,49 @@ void StackLayoutGenerator::stitchConditionalJumps(CFG::BasicBlock const& _block)
 			},
 			[&](CFG::BasicBlock::FunctionReturn const&)	{},
 			[&](CFG::BasicBlock::Terminated const&) { },
+			[&](CFG::BasicBlock::Switch const& _switch)
+			{
+				Stack exitLayout = info.exitLayout;
+				// The last block must have produced the switch expression at the stack top.
+				yulAssert(!exitLayout.empty(), "");
+				yulAssert(exitLayout.back() == _switch.switchExpr, "");
+				// The switch expression is consumed by the switch.
```

Comment:
Given my other comment, I think it's easier to generate layouts in which each switch case still *has* a copy of the switch expression in its entry layout - since that's how you'll actually end up jumping to them.
Not sure if this is the only place that needs adjustments due to that - but once you adjust the ``StackLayoutGeneratorTest``s you should be able to confirm if you get layouts that have the temporary switch expression slot as entires of the switch cases.